### DISCUSSION TITLE
Integration with llama Index?

### DISCUSSION THREAD

**Discussion by jacobwjs** [outlines-dev/outlines#697] [Q&A]
I was hoping Outlines could be a solution to problems with structured output with llama Index using open-source models (e.g. function calling, agents, etc.).  I came across https://github.com/outlines-dev/outlines/pull/380#issue-2001067539 and haven't found anything that made it's way into main. I wasn't able to find anything on the llama Index side either.

Is there anything planned, or a working example of integrating the two frameworks?

**Comment by rlouf** [comment]
> Is there anything planned, or a working example of integrating the two frameworks?

Not yet, but I'm sure an example using both frameworks would have a very broad appeal. Would you like to take a stab at it?

**Comment by jacobwjs** [comment]
I doubt I have the time for this in the next couple of weeks, but from a user perspective I don't think coming up with a mish-mash example here is the right path.

It seems "opening a PR on the LlamaIndex repo to add Outlines as a model provider" as you suggest here https://github.com/outlines-dev/outlines/pull/380#issuecomment-1818289961 get's the widest adoption with minimal effort and the best use case of Outlines.

Outlines models fit nicely into llama-Index's PydanticProgram and OutputParsers with thin wrappers, and can easily wrapped as a CustomLLM. This of course all from llama-Index side.

What are your thoughts?

**Reply by rlouf** [reply]
I think you are absolutely right!
